### ISSUE TITLE
Page not found on Edit Taxonomies

### ISSUE THREAD

**Issue by sigmundpetersen** [openfoodfoundation/openfoodnetwork#3178]
## Description

When trying to edit a Taxonomy you get 'Page not found' error

## Expected Behavior

Should enter the Edit page

## Actual Behavior

Page not found error

## Steps to Reproduce

1. In the BackOffice, Configuration -> Taxonomies
2. Right click a Taconomy and click Edit
3. Slug

## Animated Gif/Screenshot

![Untitled.png](https://images.zenhubusercontent.com/557dc8f44f7a08fbb7f338ff/29d0c0ff-48e6-4178-881c-3ce13ffd600b)

## Context

Cannot edit taxonomy, upload icons etc

## Severity

Don't know

## Your Environment

* Version used: v1.23.0
* Browser name and version: Firefox 63
* Operating System and version (desktop or mobile): Windows 10

**Comment by sstead** [comment]
On Aus staging and production...
I didn't get the same menu that @sigmundpetersen saw when I right clicked. I could successfully edit the taxonomy by clicking the edit button on the right.
![image](https://user-images.githubusercontent.com/10843056/49622770-fa4b4580-fa1e-11e8-8b96-e8619a88bff8.png)

**Comment by sigmundpetersen** [comment]
Okay I see now that it's actually in the Edit sub-menu itself the screenshot is from. There you get the right click menu. 

The original use case that didn't work was uploading icons to taxonomies. Where is that supposed to be done? Is this the place? @sstead

**Comment by sigmundpetersen** [comment]
Ping @NickWeir63 on this as well, he was trying to upload icons.

**Comment by sstead** [comment]
On Aus production our categories are all housed under one Taxonomy. I can't see anywhere to upload icons, so I imagine that's something a dev needs to do.
![image](https://user-images.githubusercontent.com/10843056/49708179-9f198d00-fc82-11e8-98b4-5c4e37f4c2e9.png)

**Comment by tschumilas** [comment]
We have not icons for taxonomies on the OFN-CAN instance either - and I get the snail when I try to select 'edit' in the sub menu.  I'd also like to figure out what the expected behaviour is re: deleting taxons.  Under what circumstances should I be able to delete a taxon?   I've made sure there are no active products with the taxon - but I still can't delete.  ??

**Comment by sigmundpetersen** [comment]
One question on setting severity is - What is the right-click-edit-action actually supposed to do? Does it open a modal? What can you edit? Has anyone ever done this? 😃 I couldn't find any guides on this. @myriamboure @RachL

**Comment by tschumilas** [comment]
Having correct and functioning taxons is essential in complex shops to help users find products.  There are many cross-instance issues it seems with taxons,  but I think our first problem, as @sigmundpetersen points out is that we don't understand what the expected behavour is,  and there is no guide/documentation on his.  Is this a spree upgrade issue I wonder?  Why does spree have right click action on these taxons if nothing happens?  Why does Spree have a tree structure (parents and children) for taxons if they aren't searchable that way?  I think we haven't even figured out what exactly the issues (more than one) are yet - before we can triage this.

**Comment by myriamboure** [comment]
See that issue @sigmundpetersen https://github.com/openfoodfoundation/openfoodnetwork/issues/2130 we explain the solution but we need someone at some point to do it ;-)
Then is it's a problem about documenting properly what is and how to do things as super admin, let's update the super admin handbook ! Let's finish it and publish it !

**Comment by tschumilas** [comment]
Thanks @myriamboure  for clarifying this issue with the taxon tags.  I know its not the issue here - but I'm wondering if you know the exptected behaviour re: editing taxons?   Try as I might -  I can't do it.  I delete any products with the taxon.  I even cancel past orders whose products might have had the taxon.  I cannot delete.  I don't open an issue because I don' t know what is the expected behaviour? Should I be able to delete non-active taxons?

**Comment by sigmundpetersen** [comment]
I think deletions are not possible because of underlying database connections. Even if you delete products or cancel orders, the connections are still there in the system. We need to implement soft-delete taxons, i.e. to hide them from view.

This issue is only regarding _editing_ taxons, so let's focus our discussion on that.

**Comment by luisramos0** [comment]
I am moving this to test ready.
When you edit a taxon it will take you to the taxon edit page (welcome to this beautiful and previously closed corner of the OFN app ❤️ ).
![Screenshot 2019-09-24 at 15 13 06](https://user-images.githubusercontent.com/1640378/65519451-d7bae800-dedd-11e9-8949-4dbb223f0e10.png)

I think a tester needs to give it a try and let us know what's good or not.

**Comment by daniellemoorhead** [comment]
I had a look at this but I'm not sure how it's supposed to work or where the changes should show up.

I was able to right click and either rename, remove, or edit the taxon (see screenshot).
<img width="364" alt="Screen Shot 2019-09-25 at 4 15 21 pm" src="https://user-images.githubusercontent.com/10949052/65574063-c0e4c600-dfaf-11e9-9795-972c9ad094ed.png">

On Katuma staging:

- I successfully renamed 'Vegetables' to be 'Vegetables!'.
- I successfully added a new taxon under the parent 'Products' category.
- I successfully removed a taxon that had no products connected to it.
- I successfully reached the 'Edit Taxon' page via the right click. <-- I think this one is the thing that wasn't working for this bug?

However:
- I attempted to remove a taxon that had products connected to it, the 'Loading' interaction displayed for a LONG time, eventually I refreshed the page and it hadn't deleted.

- I added an image to the 'Vegetables!' taxon, however this image didn't display on the shop filters section. I think maybe this is bug #2130, which hasn't been fixed yet.

@luisramos0 given that we can access the edit taxon page I assume that was the clear scope for this, so I'm calling this fixed. Tis a shame that the icons don't appear, but there's an existing open bug for that so 🤷‍♀
